networks/p2p/server.go

```diff
@@ -571,7 +571,7 @@ func (srv *MultiChannelServer) setupConn(c *conn, flags connFlag, dialDest *disc
 	}
 	c.caps, c.name, c.multiChannel = phs.Caps, phs.Name, phs.Multichannel
 
-	if c.multiChannel && dialDest != nil && (dialDest.TCPs == nil || len(dialDest.TCPs) == 0) && len(phs.ListenPort) == 2 {
+	if c.multiChannel && dialDest != nil && (dialDest.TCPs == nil || len(dialDest.TCPs) < 2) && len(phs.ListenPort) == 2 {
```

Comment:
```suggestion
	if c.multiChannel && dialDest != nil && (dialDest.TCPs == nil || len(dialDest.TCPs) < 2) && len(dialDest.TCPs) < len(phs.ListenPort) {
```
Please, consider this change.

Reply:
I think checking `(dialDest.TCPs == nil || len(dialDest.TCPs) < 2)` is not necessary.
```suggestion
	if c.multiChannel && dialDest != nil && len(dialDest.TCPs) < len(phs.ListenPort) {
```